core/src/datayoga_core/blocks/azure/read_event_hub/block.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+import logging
+from typing import AsyncGenerator, List, Optional
+
+import orjson
+from azure.eventhub import EventData, PartitionContext
+from azure.eventhub.aio import EventHubConsumerClient
+from azure.eventhub.extensions.checkpointstoreblobaio import \
+    BlobCheckpointStore
+from datayoga_core.context import Context
+from datayoga_core.producer import Message
+from datayoga_core.producer import Producer as DyProducer
+
+logger = logging.getLogger("dy")
+
+
+class Block(DyProducer):
+    """
+    Azure Event Hub block for reading events.
+    """
+
+    def init(self, context: Optional[Context] = None):
+        """
+        Initializes the block.
+
+        Args:
+            context (Context, optional): The block context. Defaults to None.
+        """
+        logger.debug(f"Initializing {self.get_block_name()}")
+
+        self.batch_size = self.properties.get("batch_size", 300)
+
+        self.consumer_client = EventHubConsumerClient.from_connection_string(
+            conn_str=self.properties["event_hub_connection_string"],
+            consumer_group=self.properties["event_hub_consumer_group_name"],
+            eventhub_name=self.properties["event_hub_name"],
+            checkpoint_store=BlobCheckpointStore.from_connection_string(
+                self.properties["checkpoint_store_connection_string"],
+                self.properties["checkpoint_store_container_name"])
+        )
+
+        self.events = {}  # Retrieved events by sequence number, used for acknowledging them once processed
+        self.messages = asyncio.Queue()
+
+    async def produce(self) -> AsyncGenerator[List[Message], None]:
+        """
+        Starts the event receiving process and yield batches of messages.
+
+        Yields:
+            AsyncGenerator[List[Message], None]: A generator of message batches.
+        """
+        logger.debug(f"Running {self.get_block_name()}")
+
+        logger.debug("Starting event receiving process")
+        asyncio.create_task(self.receive_batch())
```

Comment:
Don't leave a task without a reference. The Eventloop only keeps weak references to tasks, so it might be collected by GC at anytime.

Need something like this:
`self._task = create_task(...)`

https://docs.python.org/3/library/asyncio-task.html?highlight=create_task#asyncio.create_task